common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -25,6 +25,8 @@
 import com.google.common.collect.Lists;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.util.DiskChecker;
```

Comment:
I specifically wanted the nodemanager to continue to work with the spark shuffle service didn't load. This is because the nodemanager can run other shuffle services -> MR, TEZ, etc.  If your cluster workload for spark is in the minority then taking the entire nodemanager down doesn't necessarily make sense since the MR and TEZ jobs could run just fine on that node.  We made it configurable so you can change that behavior, just turn the config to enable it.